src/database/type-transforms.tsx

```diff
@@ -109,3 +111,67 @@ export const buildUser = (document: firebase.firestore.DocumentData): User => {
 
   return { ...pick(data, ["name", "email"]), id: document.id };
 };
+
+const buildCovidCase = (
+  doc: firebase.firestore.DocumentData,
+): ShadowCovidCase => {
+  const data = doc.data();
+
+  const observedAt = parse(doc.id, "yyyy-MM-dd", startOfToday());
+
+  return {
+    observedAt,
+    ...data,
+  };
+};
+
+export const buildShadowFacility = (
+  facilityDocument: firebase.firestore.DocumentData,
+  facilityCovidCaseDocuments: firebase.firestore.DocumentData[],
+): ShadowFacility => {
+  const data = facilityDocument.data();
+
+  // convert all known timestamps to dates
+  const toSimpleTimeseries = (record: {
+    date: firebase.firestore.Timestamp;
+    value: number;
+  }) => ({
+    date: timestampToDate(record.date),
+    value: record.value,
+  });
+
+  let {
+    state,
+    canonicalName,
+    facilityType,
+    capacity,
+    population,
+    ...other
+  } = data;
+
+  // cast known types
+  state = String(state);
+  canonicalName = String(canonicalName);
+  facilityType = String(facilityType);
+  // if these are not arrays then unfortunately they are garbage
```

Comment:
What would they be otherwise, and should we be throwing errors if the types are unexpected?

Reply:
there's not really a known reason why they would be something else ... this is just kind of a last-ditch effort to return the proper type if the Firestore document has been mangled somehow. We are not really doing this anywhere else, so it may be more trouble than it's worth? Or maybe, as you say, the more correct solution would be to throw an error (or to simply drop this document from the result set and maybe return an error that can be consumed alongside the valid data ... we don't really have a precedent here other than "assume everything is always fine" so I'm open to whatever